src/sbvr-api/permissions.ts

```diff
@@ -950,18 +950,21 @@ const checkApiKey = Promise.method((req: PermissionReq, apiKey: string) => {
 	if (apiKey == null || req.apiKey != null) {
 		return;
 	}
-	return getApiKeyPermissions(apiKey)
-		.catch(err => {
+	return Promise.join(
+		getApiKeyPermissions(apiKey).catch(err => {
 			console.warn('Error with API key:', err);
-			// Ignore errors getting the api key and just use an empty permissions object
+			// Ignore errors getting the api key and just use an empty permissions object.
 			return [];
-		})
-		.then(permissions => {
+		}),
+		getApiKeyActorId(apiKey),
```

Comment:
This will now throw instead @roman-mazur which is also a breaking change. I think to be backwards compatible it needs to be:
```ts
		getApiKeyPermissions(apiKey)
		.catch(err => {
			console.warn('Error with API key:', err);
			// Ignore errors getting the api key and just use an empty permissions object.
			return [];
		})
		.then(permissions => {
			Promise.try(() => {
				if (permissions.length > 0) {
					getApiKeyActorId(apiKey)
				}
			}).then(actor => {
				req.apiKey = {
					key: apiKey,
					permissions,
					actor,
				};
			})
		})

Reply:
@Page- sorry, but I'm lost
Where does it throw?
`getApiKeyActorId` is a function that returns a promise.
And why do we need a check for `permissions.length > 0`? Is it for optimization? But how can we make actor non-optional later in such case?

Reply:
Actor should be non-optional in practice, shouldn't it? Otherwise, the is no point in having apiKey set on a request at all.

Reply:
Please take another look.